examples/python/gui_demo.py

```diff
@@ -574,6 +582,48 @@ def on_tab_change(self, event):
     def current_tab(self):
         return DisplayType.from_tab_index(self.nb.index('current')) if self.nb is not None else DisplayType.UNSPECIFIED
 
+    def handle_start_update(self):
+        selected_item = treeview_get_first_selection(self.tree)
+        if selected_item:
+            self.begin_update_on_node_and_children(selected_item)
+            self.change_node('red')
+            self.tree_update(self.context.selected_node)
+
+    def handle_stop_update(self):
+        selected_item = treeview_get_first_selection(self.tree)
+        if selected_item:
+            self.end_update_on_node_and_children(selected_item)
+            self.change_node('red')
+            self.tree_update(self.context.selected_node)
+
+    def begin_update_on_node_and_children(self, node):
+        node_obj = find_component(node, self.context.instance)
+        node_obj = daq.IPropertyObject.cast_from(node_obj)
+        node_obj.begin_update()
+
+    def end_update_on_node_and_children(self, node):
+        node_obj = find_component(node, self.context.instance)
+        node_obj = daq.IPropertyObject.cast_from(node_obj)
+        node_obj.end_update()
+
+    def change_node(self, color):
+        for node in self.tree.get_children():
+            self._change_node_recursive(node, color)
+
+    def _change_node_recursive(self, node, color):
+        node_obj = find_component(node, self.context.instance)
+        node_text = self.get_string_from_component(node_obj.name)
+        if daq.IPropertyObject.can_cast_from(node_obj):
```

Comment:
I don't think we need to cast. Every object in the tree should descend from the property object.